File: frontend/types/growth.ts
export interface Growth {
    id: number;
    user_id: number;
    baby_id: number;
    date: string; // ISO 8601 (date)
    weight: number | null; // in grams
    height: number | null; // in cm
    head_circumference: number | null; // in cm
    notes: string | null;
}

export interface GrowthCreate {
    baby_id: number;
    date: string;
    weight?: number;
    height?: number;
    head_circumference?: number;
    notes?: string;
}

File: app/models/growth.py
from sqlalchemy import Column, Integer, String, ForeignKey, Date, Float, Index
from .base import Base, SoftDeleteMixin


class Growth(Base, SoftDeleteMixin):
    __tablename__ = "growths"

    id = Column(Integer, primary_key=True, autoincrement=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    date = Column(Date, nullable=False, index=True)
    weight = Column(Integer, nullable=True)  # in grams
    height = Column(Float, nullable=True)  # in cm
    head_circumference = Column(Float, nullable=True)  # in cm
    notes = Column(String, nullable=True)
    baby_id = Column(Integer, ForeignKey("babies.id"), nullable=False)

    __table_args__ = (
        Index("idx_growth_baby_date", "baby_id", "date"),
        Index("ix_growths_baby_id_is_deleted", "baby_id", "is_deleted"),
    )
